tests/encoders/base.py

```diff
@@ -0,0 +1,208 @@
+import unittest
```

Comment:
Why use unittest instead of pytest? unittest is more or less obsolete

Reply:
Yeah, but I like the idea of test inheritance, don't know if we can use it with pytest. But we still use pytest to collect and run tests